### PR TITLE
FIx TMVA Deep Learning  for builds without IMT and/or BLAS +  tutorials fixes

### PR DESCRIPTION
**Pull request by lmoneta**
- Fix the computation in case of imt or blas is not available.  

Use still TCpuMatrix class (TCpu architecture) but use TMatrix for matrix multiplication instead of BLAS\
  Only dense layers architectures are supported in this case

  - Fix tutorials for the no-imt case

-   Use Higgs data file from root.cern.ch

### CODE REVIEW

File: tmva/tmva/src/MethodDL.cxx [hunk 1]
```diff
@@ -1672,36 +1668,22 @@ void MethodDL::Train()
...
-      return;
 #endif
+      TrainDeepNet<DNN::TCpu<ScalarImpl_t> >();
+      return;
    } else if (this->GetArchitectureString() == "STANDARD") {
-      Log() << kINFO << "Start of deep neural network training on the STANDARD architecture" << Endl << Endl;
-#if HAVE_REFERENCE
-      TrainDeepNet<DNN::TReference<ScalarImpl_t> >();
-#endif
+      Log() << kFATAL << "The STANDARD architecture is not supported anymore !" << Endl << Endl;
```
Comment (Axel-Naumann) [inline tmva/tmva/src/MethodDL.cxx:1678]: "Please use..."?

File: tmva/tmva/src/MethodDL.cxx [hunk 2]
```diff
@@ -2054,20 +2036,13 @@ std::vector<Double_t> MethodDL::GetMvaValues(Long64_t firstEvt, Long64_t lastEvt
...
    }
-   Log() << kINFO << "ERROR:  STANDARD architecture  is not supported anymore for MethodDL ! "
-         << Endl << Endl;
-// #if HAVE_REFERENCE
-//    return PredictDeepNet<DNN::TReference<ScalarImpl_t> >(firstEvt, lastEvt, batchSize, logProgress);
-// #else
-   return std::vector<Double_t>(nEvents,TMath::QuietNaN());
-//#endif
-
+   Log() << kERROR << "STANDARD architecture  is deprecated for MethodDL ! " << Endl << Endl;
```
Comment (Axel-Naumann) [inline tmva/tmva/src/MethodDL.cxx:2043]: That doesn't look like deprecation - it's now unavailable, right? "Deprecation" means "will be removed in a future version", where here it's just gone without prior warning. Not really nice towards our users, unless it's unused or broken?
It'd be also good to state an alternative for people to use. Or even better, to use a fall-back.

File: tmva/tmva/test/DNN/CMakeLists.txt [hunk 3]
```diff
@@ -190,7 +192,7 @@ endif ()
     LIBRARIES ${Libraries})
   ROOT_ADD_TEST(TMVA-DNN-Regression-Cpu COMMAND testRegressionCpu)
 
-endif ()
+####endif()
```
Comment (Axel-Naumann) [inline tmva/tmva/test/DNN/CMakeLists.txt:195]: Would one `#` be enough?

**Comment by lmoneta** [comment]
The option Standard was before a complete implementation of the dense layer using the TMatrix classes. It was mainly for testing, and as far as I know not used. 
A replacements exists now, and it is used when TMVA is built without BLAS or IMT support. But before this PR it was buggy and for this reason the no-imt nightly builds were failing. 

I will apply the changes as you suggest above before merging this PR

**Pull request merged**